alacritty_terminal/src/renderer/rects.rs

```diff
@@ -90,33 +91,19 @@ impl RenderLines {
             .collect()
     }
 
-    /// Update the stored lines with the next cell info.
-    pub fn update(&mut self, cell: &RenderableCell) {
+    /// Update the stored lines with the next text_run info.
+    pub fn update(&mut self, text_run: &TextRun) {
```

Comment:
Currently it seems like this method doesn't have a ton of logic in it anymore.

I don't think we need a dedicated struct for this anymore, we already have a collection of text runs and those are already separated by flags, so we've got all the info we need.

I think we could just make a method on render rect which takes a text run as parameter and then we'll just create all rects by iterating over all text runs, filtering by flags and then creating render rects for it.

Does that make sense to you? I just think the middle man isn't super necessary anymore, we already have the data structure ready as text runs, so all we need is a single method to convert them.

Reply:
I missed this one entirely, sorry about that. I wholeheartedly agree, I've removed the RenderLines struct and replaced it with a `from_text_run` method on RenderRect. The lines local in draw is replaced with a Vec of RenderRects.